game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1039,49 +1035,60 @@ private void pushFightLoopOnStack() {
    * It is allowed for an IExecutable to add other IExecutables to the stack. If you read the code
    * in linear order, ignore wrapping stuff in anonymous IExecutables, then the code can be read as
    * it will execute. The steps are added to the stack and then reversed at the end.
-   *
-   * <p>Save Game Compatibility Note:
-   *
-   * <p>Because of saved game compatibility issues, the original steps are left behind as inner
-   * anonymous classes. The reason for this is that their class name is defined by the order in
-   * which they are defined. As an example, the first inner anonymous class is MustFightBattle$0 and
-   * the next one is MustFightBattle$1 and so on. When a saved game is deserialized, it will match
-   * the step by the class name and so if the order of inner anonymous classes change, it will not
-   * deserialize. So even though these old steps aren't being added to the steps array, they are
-   * still needed. They can be safely removed once save compatibility can be broken.
    */
   @VisibleForTesting
   public List<IExecutable> getBattleExecutables() {
-    final List<IExecutable> steps = new ArrayList<>();
-    addFightStartSteps(steps);
-    addFightSteps(steps);
-    addCheckEndBattleAndRetreatingSteps(steps);
-    return steps;
-  }
-
-  private void addFightStartSteps(final List<IExecutable> steps) {
     if (offensiveAa == null) {
       updateOffensiveAaUnits();
     }
     if (defendingAa == null) {
       updateDefendingAaUnits();
     }
-    final List<BattleStep> startSteps =
+    final List<BattleStep> battleSteps =
         List.of(
             new OffensiveAaFire(this, this),
             new DefensiveAaFire(this, this),
             new ClearAaCasualties(this, this),
             new NavalBombardment(this, this),
             new RemoveNonCombatants(this),
             new LandParatroopers(this, this),
-            new MarkNoMovementLeft(this, this));
-    steps.addAll(
-        startSteps.stream()
+            new MarkNoMovementLeft(this, this),
+            new OffensiveSubsRetreat(this, this),
+            new DefensiveSubsRetreat(this, this),
+            new OffensiveFirstStrike(this, this),
+            new DefensiveFirstStrike(this, this),
+            new ClearFirstStrikeCasualties(this, this),
+            new OffensiveGeneral(this, this),
+            new DefensiveGeneral(this, this),
+            new RemoveUnprotectedUnits(this, this),
+            new RemoveFirstStrikeSuicide(this, this),
+            new SubmergeSubsVsOnlyAirStep(this, this));
+    final List<IExecutable> steps =
+        battleSteps.stream()
             .sorted(Comparator.comparing(BattleStep::getOrder))
-            .collect(Collectors.toList()));
+            // *_AFTER_BATTLE order occurs in addCheckEndBattleAndRetreatingSteps()
```

Comment:
The significance of this comment is not obvious to me. Should this comment be clarified to explain why it's important, or perhaps removed?